overlord/snapstate/handlers.go

```diff
@@ -1926,24 +1925,31 @@ func setMigrationFlagsinState(snapst *SnapState, snapsup *SnapSetup) {
 	}
 }
 
-// maybeRestart will schedule a reboot or restart as needed for the
-// just linked snap with info if it's a core or snapd or kernel snap.
-func (m *SnapManager) maybeRestart(t *state.Task, info *snap.Info, rebootInfo boot.RebootInfo) {
+// maybeRestart carries information to decide whether a restart of some form
+// is required.
+type maybeRestart struct {
+	info *snap.Info
+	boot.RebootInfo
+}
```

Comment:
`maybeRestart` sounds like a function, I generally wouldn't call a struct with a name containing `maybe`. I suggest to call it `restartInfo` or `restartContext`.

Reply:
those name are bit too generic, also they lose the connection that non-nil pointers to these struct are used to signal that a restart needs to be considered. I went with restartPossibility which at least is a noun and also expanded the  doc comment.